src/renderers/shaders/ShaderChunk/emissivemap_pars_fragment.glsl

```diff
@@ -2,4 +2,10 @@
 
 	uniform sampler2D emissiveMap;
 
+	vec4 emissiveMapTexelToLinear( vec4 value ) {
+		#define MACRO_DECODE EMISSIVEMAP_ENCODING
+			#include <encoding_template>
```

Comment:
If we had a better preprocessor I believe we could.  I really struggled with the limitations of glsl in this case to avoid using up precious uniforms for the encoding selection.

BTW with this new preprocessor that supports includes we can gut ShaderLib. :)  What I suggest is we refactor the large single-level WebGL/shaders/chunks directory so that we have a directory for each shader (with a frag/vert shader each) and then a directory (at least) of shared chunks.

Both of these above changes can be done after this is merged though. :)  Also I believe that @tschw is the compiler expert, although I am not suggesting he has time to look into this -- I sort of fake being a compiler guy.